store/sqlstore/group_store.go

```diff
@@ -963,18 +963,21 @@ func (s *SqlGroupStore) GetGroupsByTeam(teamId string, opts model.GroupSearchOpt
 func (s *SqlGroupStore) GetGroups(page, perPage int, opts model.GroupSearchOpts) ([]*model.Group, *model.AppError) {
 	var groups []*model.Group
 
-	groupsQuery := s.getQueryBuilder().Select("g.*").From("UserGroups g").Limit(uint64(perPage)).Offset(uint64(page * perPage)).OrderBy("g.DisplayName")
+	groupsQuery := s.getQueryBuilder().Select("g.*").From("UserGroups g")
 
 	if opts.IncludeMemberCount {
 		groupsQuery = s.getQueryBuilder().
 			Select("g.*, coalesce(Members.MemberCount, 0) AS MemberCount").
 			From("UserGroups g").
-			LeftJoin("(SELECT GroupMembers.GroupId, COUNT(*) AS MemberCount FROM GroupMembers LEFT JOIN Users ON Users.Id = GroupMembers.UserId WHERE GroupMembers.DeleteAt = 0 AND Users.DeleteAt = 0 GROUP BY GroupId) AS Members ON Members.GroupId = g.Id").
-			Limit(uint64(perPage)).
-			Offset(uint64(page * perPage)).
-			OrderBy("g.DisplayName")
+			LeftJoin("(SELECT GroupMembers.GroupId, COUNT(*) AS MemberCount FROM GroupMembers LEFT JOIN Users ON Users.Id = GroupMembers.UserId WHERE GroupMembers.DeleteAt = 0 AND Users.DeleteAt = 0 GROUP BY GroupId) AS Members ON Members.GroupId = g.Id")
```

Comment:
Can we also extract out the From Usergroups g? Right before the Left join? I noticed it is repeated in both.

Reply:
Nope because the left join needs the `From("UserGroups g") in order to be a valid query. I think that having that one line in both blocks is fine

Reply:
Good point. I didn't catch that. Thanks!

Reply:
looks like I was wrong order does not matter for the query builder to work properly - applied this change in the latest commit